### PR TITLE
Added Resource and Data Source Files for Access/Bypass Codes , Named Lists , Network Lists

### PR DESCRIPTION
**Pull request by unasra**

### CODE REVIEW

File: internal/service/fw/api_access_codes_resource.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+		Body(*data.Expand(ctx, &resp.Diagnostics)).
+		Execute()
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to create AccessCodes, got error: %s", err))
+		return
+	}
+
+	res := apiRes.GetResults()
+	data.Flatten(ctx, &res, &resp.Diagnostics)
+	//resp.State.SetAttribute(ctx, path.Root("id"), data.AccessKey.ValueString())
```
Comment (mathewab) [inline internal/service/fw/api_access_codes_resource.go:81]: to be removed?

File: internal/service/fw/api_access_codes_resource_test.go [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccAccessCodesName(name1),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAccessCodesExists(context.Background(), resourceName, &v1),
+					resource.TestCheckResourceAttr(resourceName, "name", name1),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.data", "terraform_test"),
```
Comment (mathewab) [inline internal/service/fw/api_access_codes_resource_test.go:81]: `data` and `type` don't have to be verified here, makes it redundant. They should be in basic, and in the test that is specific to that field.

File: internal/service/fw/api_access_codes_resource_test.go [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccAccessCodesActivation(name, time.Now().UTC().Format(time.RFC3339)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAccessCodesExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.data", "terraform_test"),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.type", "custom_list"),
+					resource.TestCheckResourceAttr(resourceName, "activation", time.Now().UTC().Format(time.RFC3339)),
```
Comment (mathewab) [inline internal/service/fw/api_access_codes_resource_test.go:117]: isn't there a chance that `time.Now()` in L112 and here might be different. You should use a variable instead,

File: internal/service/fw/api_access_codes_resource_test.go [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccAccessCodesActivation(name, time.Now().UTC().Add(time.Minute*10).Format(time.RFC3339)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAccessCodesExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.data", "terraform_test"),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.type", "custom_list"),
+					resource.TestCheckResourceAttr(resourceName, "activation", time.Now().UTC().Add(time.Minute*10).Format(time.RFC3339)),
```
Comment (mathewab) [inline internal/service/fw/api_access_codes_resource_test.go:127]: Same comment as above.

File: internal/service/fw/api_access_codes_resource_test.go [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccAccessCodesExpiration(name, time.Now().UTC().Add(time.Hour).Format(time.RFC3339)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAccessCodesExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.data", "terraform_test"),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.type", "custom_list"),
+					resource.TestCheckResourceAttr(resourceName, "expiration", time.Now().UTC().Add(time.Hour).Format(time.RFC3339)),
```
Comment (mathewab) [inline internal/service/fw/api_access_codes_resource_test.go:185]: same comment as above

File: docs/resources/td_access_code.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Optional
+
+- `activation` (String) The time when the Bypass Code object was activated.
+- `description` (String) The brief description for an access code.
+- `expiration` (String) The time when the Bypass Code object was expired.
```
Comment (mathewab) [inline docs/resources/td_access_code.md:22]: `was expired` sounds incorrect :)

File: internal/service/fw/model_atcfw_access_code_rule.go [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+var AtcfwAccessCodeRuleAttrTypes = map[string]attr.Type{
+	"action":        types.StringType,
+	"data":          types.StringType,
+	"description":   types.StringType,
+	"redirect_name": types.StringType,
+	"type":          types.StringType,
+}
+
+var AtcfwAccessCodeRuleResourceSchemaAttributes = map[string]schema.Attribute{
```
Comment (mathewab) [inline internal/service/fw/model_atcfw_access_code_rule.go:33]: is this used?

File: internal/service/fw/model_atcfw_named_list.go [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+		NestedObject: schema.NestedAttributeObject{
+			Attributes: AtcfwItemStructsResourceSchemaAttributes,
+		},
+		Optional:            true,
+		MarkdownDescription: "The List of ItemStructs structure which contains the item and its description",
+	},
+	"name": schema.StringAttribute{
+		Optional: true,
+		PlanModifiers: []planmodifier.String{
+			stringplanmodifier.RequiresReplaceIfConfigured(),
```
Comment (mathewab) [inline internal/service/fw/model_atcfw_named_list.go:97]: you are not allowed to update name?

File: examples/resources/bloxone_td_access_codes/resource.tf [hunk 9]
```diff
@@ -0,0 +1,17 @@
+resource "bloxone_td_access_code" "example" {
+  name       = "example_access_code"
+  activation = timestamp()
+  expiration = timeadd(timestamp(), "24h")
+  rules = [
+    {
+      action        = "",
```
[the remark this answers is not in the document]
Comment (unasra) [reply examples/resources/bloxone_td_access_codes/resource.tf:7]: Some fields require empty string to be passed. I have removed here and added default values.

File: internal/flex/flex.go [hunk 10]
```diff
@@ -87,6 +94,15 @@ func ExpandFrameworkListString(ctx context.Context, tfList types.List, diags *di
 	return data
 }
 
+func ExpandFrameworkListInt64(ctx context.Context, tfList types.List, diags *diag.Diagnostics) []int32 {
```
[the remark this answers is not in the document]
Comment (unasra) [reply internal/flex/flex.go:97]: Done

File: internal/service/fw/api_access_codes_resource_test.go [hunk 11]
```diff
@@ -0,0 +1,408 @@
...
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccAccessCodesBasicConfig(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAccessCodesExists(context.Background(), resourceName, &v),
```
[the remark this answers is not in the document]
Comment (unasra) [reply internal/service/fw/api_access_codes_resource_test.go:35]: Done

File: internal/service/fw/api_access_codes_resource.go (hunk 1 above)
Comment (unasra) [reply internal/service/fw/api_access_codes_resource.go:81]: Done

File: internal/service/fw/api_access_codes_resource_test.go (hunk 2 above)
Comment (unasra) [reply internal/service/fw/api_access_codes_resource_test.go:81]: Removed . Similar fields have been removed from other object too

File: internal/service/fw/api_access_codes_resource_test.go (hunk 3 above)
Comment (unasra) [reply internal/service/fw/api_access_codes_resource_test.go:117]: Added a variable instead.

File: internal/service/fw/api_access_codes_resource_test.go (hunk 4 above)
Comment (unasra) [reply internal/service/fw/api_access_codes_resource_test.go:127]: Resolved

File: internal/service/fw/api_access_codes_resource_test.go (hunk 5 above)
Comment (unasra) [reply internal/service/fw/api_access_codes_resource_test.go:185]: Resolved

File: docs/resources/td_access_code.md (hunk 6 above)
Comment (unasra) [reply docs/resources/td_access_code.md:22]: Fixed

File: internal/service/fw/model_atcfw_named_list.go (hunk 8 above)
Comment (unasra) [reply internal/service/fw/model_atcfw_named_list.go:97]: Removed the planmodifier

Comment (mathewab) [reply internal/service/fw/model_atcfw_named_list.go:97]: This one still seems to be here.

File: examples/resources/bloxone_td_access_codes/resource.tf (hunk 9 above)
[the remark this answers is not in the document]
Comment (mathewab) [reply examples/resources/bloxone_td_access_codes/resource.tf:7]: you also have to test whether these fields can be unset. For e.g. if it was "xyz" before, passing "" would actually unset it, and not just ignore the change.

File: internal/service/fw/model_atcfw_access_code_rule.go [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
```
Comment (mathewab) [inline internal/service/fw/model_atcfw_access_code_rule.go:34]: Can you please add description for these fields?

File: docs/resources/td_named_list.md
Comment (mathewab) [inline docs/resources/td_named_list.md:1]: Are there no required fields for this resource?

File: docs/resources/td_access_code.md
Comment (mathewab) [inline docs/resources/td_access_code.md:1]: Are there no required fields for this resource? Also there is no example for this.

File: internal/service/fw/model_atcfw_access_code_rule.go (hunk 12 above)
Comment (unasra) [reply internal/service/fw/model_atcfw_access_code_rule.go:34]: Added

File: docs/resources/td_network_list.md
[the remark this answers is not in the document]
Comment (unasra) [reply docs/resources/td_network_list.md:1]: Fixed

File: docs/resources/td_named_list.md
Comment (unasra) [reply docs/resources/td_named_list.md:1]: Fixed

File: internal/service/fw/model_atcfw_named_list.go (hunk 8 above)
Comment (unasra) [reply internal/service/fw/model_atcfw_named_list.go]: Removed for sure this time

**Review by mathewab — APPROVED** [review]

**Pull request merged**